sql: fix SCRUB to work on TEMP TABLE

otan opened the pull request:
Release note (bug fix): Fix a bug where EXPERIMENTAL SCRUB did not work
on temporary tables.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47443)
<!-- Reviewable:end -->

otan committed:
sql: fix SCRUB to work on TEMP TABLE

Release note (bug fix): Fix a bug where EXPERIMENTAL SCRUB did not work
on temporary tables.

rohany commented on pkg/sql/scrub_constraint.go:
```diff
@@ -96,8 +97,26 @@ func (o *sqlCheckConstraintCheckOperation) Start(params runParams) error {
+	if err != nil {
+		return err
+	}
+	if isTempSchema {
```
shouldn't we just always put the temp schema in the search path here? Or rather, why isn't the temp schema in the search path for this query?

otan replied:
hmm, so for some reason the session data is not set for the internal executor here.

otan replied:
i'm going to close this as i think the fix is deeper rooted.
